libminifi/test/unit/ChecksumCalculatorTests.cpp

```diff
@@ -0,0 +1,101 @@
+  TestController test_controller;
+  std::string test_dir = utils::createTempDir(&test_controller);
+  std::string file_location = utils::putFileToDir(test_dir, "simple.txt", "one line of text\r\n");
+
+  utils::ChecksumCalculator checksum_calculator;
+  checksum_calculator.setFileLocation(file_location);
+  REQUIRE(checksum_calculator.getChecksum() == "94fc46c62ef6cc5b45cbad9fd53116cfb15a80960a9b311c1c27e5b5265ad4b4");
+}
+
+TEST_CASE("The checksum can be reset and recomputed", "[ChecksumCalculator]") {
+  TestController test_controller;
+  std::string test_dir = utils::createTempDir(&test_controller);
+  std::string file_location = utils::putFileToDir(test_dir, "simple.txt", "one line of text\n");
+
+  utils::ChecksumCalculator checksum_calculator;
+  checksum_calculator.setFileLocation(file_location);
+  REQUIRE(checksum_calculator.getChecksum() == CHECKSUM_FOR_ONE_LINE_OF_TEXT);
+
+  std::ofstream append_to_file(file_location, std::ios::binary | std::ios::app);
+  append_to_file << "another line of text\n";
+  append_to_file.close();
+
+  REQUIRE(checksum_calculator.getChecksum() == CHECKSUM_FOR_ONE_LINE_OF_TEXT);  // not updated, needs to be notified
+
+  checksum_calculator.invalidateChecksum();
+  REQUIRE(checksum_calculator.getChecksum() == CHECKSUM_FOR_TWO_LINES_OF_TEXT);  // now it is updated
+}
+
+TEST_CASE("Checksums can be computed for binary (eg. encrypted) files, too", "[ChecksumCalculator]") {
+  TestController test_controller;
+  std::string test_dir = utils::createTempDir(&test_controller);
+  std::string binary_data(size_t{256}, '\0');
+  std::iota(binary_data.begin(), binary_data.end(), 'x');
+  std::string file_location = utils::putFileToDir(test_dir, "simple.txt", binary_data);
+
+  utils::ChecksumCalculator checksum_calculator;
+  checksum_calculator.setFileLocation(file_location);
+  REQUIRE(checksum_calculator.getChecksum() == "bdec77160c394c067419735de757e4daa1c4679ea45e82a33fa8f706eed87709");
+}
+
+TEST_CASE("The agent identifier is excluded from the checksum", "[ChecksumCalculator]") {
+  TestController test_controller;
+  std::string test_dir = utils::createTempDir(&test_controller);
+  std::string file_location_1 = utils::putFileToDir(test_dir, "agent_one.txt",
+      "nifi.c2.agent.class=Test\n"
+      "nifi.c2.agent.identifier=Test-111\n"
+      "nifi.c2.agent.heartbeat.period=10000\n");
+  std::string file_location_2 = utils::putFileToDir(test_dir, "agent_two.txt",
+      "nifi.c2.agent.class=Test\n"
+      "nifi.c2.agent.identifier=Test-222\n"
+      "nifi.c2.agent.heartbeat.period=10000\n");
+
+  utils::ChecksumCalculator checksum_calculator_1;
+  checksum_calculator_1.setFileLocation(file_location_1);
+  utils::ChecksumCalculator checksum_calculator_2;
+  checksum_calculator_2.setFileLocation(file_location_2);
+
+  REQUIRE(checksum_calculator_1.getChecksum() == checksum_calculator_2.getChecksum());
+}
```

Comment:
Some additional simple tests could be added for error cases of `getChecksum` and `getFileName` if the file location is not set.

Reply:
added tests of the error cases in https://github.com/apache/nifi-minifi-cpp/pull/1123/commits/f6868eb85a9f286c6229984d42e03ecd95dd0e86